packages/react-storage/src/components/StorageBrowser/views/hooks/usePaginate.ts

```diff
@@ -25,6 +26,10 @@ export const usePaginate = ({
     setCurrentPage(1);
   }).current;
 
+  const isInitialPage = currentPage === 1;
+  const start = isInitialPage ? 0 : (currentPage - 1) * pageSize;
+  const end = isInitialPage ? pageSize : currentPage * pageSize;
```

Comment:
Since `start` and `end` are resolved with values already included in the memoized return dep array, might make sense to resolve them inside the return value?